rizzino_a07b.js

```diff
@@ -0,0 +1,123 @@
+buildFoodsList (badFoods, "bread", "pasta", "pizza", "fries", "candy");          
+//buildFoodsList function has 5 items to be passed to the badFoods array.
+
+standardInput.on('data', function (text) {
+    //This line will remove line breaks 
+    //\n\r in window, \r in macOS
+    text = text.replace('\n', '').replace('\r', '');
+    //Step 4 Begin
+    if (text == "Q"||"q")
+    //if statement for text equal to "Q" will clear the screen, say "Bye", and end the program.
+    {
+        clearScreen();
+        console.log("Bye");
+        process.exit();
+    } else 
+    //else statment for when text is something other than "Q" will clear the screen and perform checkItem function.
+    {
+        clearScreen();
+        checkItem(text,badFoods);
+        if (checkItem() == false)
+        //if checkItem function returns false the text will be formated in the formatItem function, then added to the groceryList array, and output message for item added to list followed by the getPrompt function.
+        {
+            formatItem(text);
+            groceryItem.push(groceryItems());
+            console.log(`Item added to list!
+
+List contains ${groceryItems.length} items`);
+getPrompt();
+        } else if (checkItem() == true) 
+        //if checkItem function returns true the text will not be added and an error message will output followed by the getPrompt function.
+        {
+            console.log(`Error: that item is not safe for your allergies.
+It has not been added to your grocery list.`);
+getPrompt();
+        } else 
+        //if another response returns from the checkItem function it will output the getPrompt function as default.
+        {
+            getPrompt();
+        }
+    }
+    
+    //Step 4 End
+});
+
+/* DEFINE YOUR FUNCTIONS BETWEEN THIS LINE */
+
+//Step 2
+
+function buildFoodsList(badFoods,...args) 
+//the buildFoodsList function is building the badFoods array with a varible number of arguments. It will then iterate through the number of arguments and add them to the badFoods array.
+{
+    function rest(...args){
+        return args;
+    }
+    console.log(args);
+    //included console log of args to check if arguments were loading correctly.
+    for (n = 0; n < args.length; n++){
+        badFoods.push(args[n]);
+    }
+    //for loop loads the args into the badFoods array
+    console.log(badFoods);
+    //included console log of badFoods to check if arguments had been loaded into array.
+    console.log(badFoods.length);
+    //included console log of badFoods.length to check if it was showing up correctly to be used in later functions.
+}
+
+function getPrompt(groceryItems) 
+//the getPrompt function will output a message that says how many items are in your groceryList and will ask for items to be entered or give you the option to quit the program.
+{
+    console.log(
+`Your grocery list contains ${groceryItems.length} items. 
+Please enter a grocery item. 
+Enter Q to quit 
+======================`);
```

Comment:
playing around with things here - you need a return on this function. So instead of console.log() in the function - store the template to a var and then return it. The main program logs it for you anyway.

Also - you need the Enter V line.